include/up-cpp/communication/Subscriber.h

```diff
@@ -43,17 +43,18 @@ struct Subscriber {
 	/// @returns * A unique_ptr to a Subscriber if the callback was
 	///            successfully registered.
 	///          * A UStatus with the appropriate failure code otherwise.
-	[[nodiscard]] utils::Expected<std::unique_ptr<Subscriber>, v1::UStatus>
+	[[nodiscard]] static utils::Expected<std::unique_ptr<Subscriber>,
+	                                     v1::UStatus>
 	subscribe(std::shared_ptr<transport::UTransport> transport,
 	          const v1::UUri& topic,
 	          transport::UTransport::ListenCallback&& callback);
 
-protected:
+public:
 	/// @brief Constructor
 	///
 	/// @param transport Transport this subscriber is connected to.
 	/// @param subscription Handle to the callback registered with UTransport.
-	Subscriber(std::shared_ptr<transport::UTransport> transport,
+	Subscriber(const std::shared_ptr<transport::UTransport>& transport,
 	           transport::UTransport::ListenHandle&& subscription);
```

Comment:
Shouldn't the constructor be protected to prevent users from calling it directly and not using the static subscribe function?

Reply:
@billpittman I also thought the same. If I make constructor as protected then I cannot invoke it from subscribe method because it is a static method. As per current design, subscribe method should be static.
@gregmedd  let us know your thoughts on this comment

Reply:
Done. Constructor is protected now

Reply:
Looks good. Thanks for helping get that figured out :+1: